core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionClient.scala

```diff
@@ -429,7 +429,7 @@ private[spark] object RestSubmissionClient {
    */
   private[rest] def filterSystemEnvironment(env: Map[String, String]): Map[String, String] = {
     env.filter { case (k, _) =>
-      (k.startsWith("SPARK_") && k != "SPARK_ENV_LOADED") || k.startsWith("MESOS_")
+      (k.startsWith("SPARK_") && k != "SPARK_ENV_LOADED" && k != "SPARK_HOME") || k.startsWith("MESOS_")
```

Comment:
we should add a comment here to explain why we need to filter out SPARK_HOME, and point to the JIRA. (though we can always do that in a separate patch so as to unblock the release)